### ISSUE TITLE
unique values with check_ping (CheckNet)

### ISSUE THREAD

**Issue by gschlad** [mickem/nscp#536]
### Issue and Steps to Reproduce

i'm currently building a new process for an end2end monitoring. i wanted to include the values of check_ping (nsclient checkNet module) to monitor the response time from remote clients to a central application server, but i don't get comprehensible values. 

```
check_nrpe -H nsclient-host -t 60 -c check_ping -a host=10.10.0.1 show-all "warn=time >45 or loss >15" "crit=time >80 or loss >30"
```

### Expected Behavior

this is like it looks like with "check_icmp" plugin from my application server to the client. I get values between 20 and 25ms:

![image](https://user-images.githubusercontent.com/25278906/34918346-69379342-f951-11e7-8dbc-4209fdeb8344.png)

### Actual Behavior

this is what it looks like with check_ping from a windows host with nsclient to the application server:

![image](https://user-images.githubusercontent.com/25278906/34918350-89e2a17c-f951-11e7-85b3-2638d9823b41.png)

WARNING: 0/1 (warning(10.10.0.1 Packet loss = 0%, RTA = 15ms))|'10.10.0.1_time'=15;10;300
WARNING: 0/1 (warning(10.10.0.1 Packet loss = 0%, RTA = 31ms))|'10.10.0.1_time'=31;10;300

mostly I only get values with exactly 15 or exactly 31ms, only very rarely it is a different value. I got the same values on different location all over the world, most of them have much more then 15ms answer time, so they should much higher and more different values. 

### Details

* NSClient++ version: 0.5.1.19 and 0.5.1.44
* OS and Version: Windows Server 2012, Windows 7 SP2, Windows 10 1709
* Checking from: Naemon
* Checking with: check_nrpe

**Comment by mintsoft** [comment]
Hmm I think the values are actually `15.625` and `31.25` which are multiples of the default windows clock resolution timer (https://docs.microsoft.com/en-us/sysinternals/downloads/clockres). I think the way that the time difference is being calculated in `nscp` needs to be switched to something high resolution

**Comment by mintsoft** [comment]
Looking at the code for `check_ping` (https://github.com/mickem/nscp/blob/da7c4b9b8f1036a0176f0ffd90070134b823d5f0/modules/CheckNet/CheckNet.cpp#L85 and https://github.com/mickem/nscp/blob/da7c4b9b8f1036a0176f0ffd90070134b823d5f0/include/net/pinger.hpp#L129) it seems to be relying on subtracting `posix_time::microsec_clock::universal_time` objects to get a duration. According to the boost docs:

> Get the local time using a sub second resolution clock. On Unix systems this is implemented using GetTimeOfDay. On most Win32 platforms it is implemented using ftime. Win32 systems often do not achieve microsecond resolution via this API. If higher resolution is critical to your application test your platform to see the achieved resolution.

That smells very much like the behaviour that in .NET if you subtract 2 DateTime objects, you only get the time different accurate to 15.625 ms intervals; using Stopwatch you get it accurate to the tick.

Alternatively, given that the timing appears to be Async it might have to wait until a threadpool is pumped before the callback is executed (and the times are measured). For example under windows with standard clock resolution settings if you do `Thread.sleep(1)` (which should sleep for 1ms) then it will actually sleep for `15.625ms`. I've been looking through the boost asio implementation and I can't find anything that makes it obvious that it's susceptable for this, but it seems very suspicious!

**Comment by mickem** [comment]
This has been fixed.
